src/middlewared/middlewared/plugins/jail.py

```diff
@@ -677,6 +698,49 @@ def get_plugin_version(self, pkg):
 
         return version
 
+    @private
+    def get_local_plugin_version(self, plugin, index_json, iocroot):
+        """
+        Checks the primary_pkg key in the INDEX with the pkg version
+        inside the jail.
+        """
+        if index_json is None:
+            return ['N/A', 'N/A']
+
+        try:
+            base_plugin = plugin.rsplit('_', 1)[0]  # May have multiple
+            primary_pkg = index_json[base_plugin]['primary_pkg']
+            version = ['N/A', 'N/A']
+
+            # Since these are plugins, we don't want to spin them up just to
+            # check a pkg, directly accessing the db is best in this case.
+            db_rows = self.read_plugin_pkg_db(
+                f'{iocroot}/jails/{plugin}/root/var/db/pkg/local.sqlite')
+
+            for row in db_rows:
+                if primary_pkg in row[1] or primary_pkg in row[2]:
+                    version = [row[3], '1']
+                    break
+        except KeyError:
+            version = ['N/A', 'N/A']
+
+        return version
+
+    @private
+    def read_plugin_pkg_db(self, db):
+        try:
+            conn = sqlite3.connect(db)
+        except sqlite3.Error as e:
+            raise CallError(e)
+
+        with conn:
+            cur = conn.cursor()
+            cur.execute('SELECT * FROM packages')
```

Comment:
That can be very inefficient, can we use WHERE clauses here to only get the exact pkg we want?